debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java

```diff
@@ -1103,6 +1104,67 @@ public void shouldReceiveChangesForTableWithoutPrimaryKey() throws Exception{
     }
 
+    @Test
+    @FixFor("DBZ-1146")
+    public void shouldReceiveChangesForReplicaIdentityFullTableWithToastedValue() throws Exception {
+        testReceiveChangesForReplicaIdentityFullTableWithToastedValue(SchemaRefreshMode.COLUMNS_DIFF_EXCLUDE_UNCHANGED_TOAST, true);
+        testReceiveChangesForReplicaIdentityFullTableWithToastedValue(SchemaRefreshMode.COLUMNS_DIFF_EXCLUDE_UNCHANGED_TOAST, false);
+    }
+
+    private void testReceiveChangesForReplicaIdentityFullTableWithToastedValue(PostgresConnectorConfig.SchemaRefreshMode mode, boolean tablesBeforeStart) throws Exception{
+        PostgresConnectorConfig config = new PostgresConnectorConfig(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.SCHEMA_REFRESH_MODE, mode)
+                .build());
+        setupRecordsProducer(config);
+
+        if (tablesBeforeStart) {
+            TestHelper.execute(
+                    "DROP TABLE IF EXISTS test_table;",
+                    "CREATE TABLE test_table (id SERIAL, not_toast int, text TEXT);",
+                    "ALTER TABLE test_table REPLICA IDENTITY FULL"
+            );
+        }
+
+        if (!tablesBeforeStart) {
+            TestHelper.execute(
+                    "DROP TABLE IF EXISTS test_table;",
+                    "CREATE TABLE test_table (id SERIAL, not_toast int, text TEXT);",
+                    "ALTER TABLE test_table REPLICA IDENTITY FULL"
+            );
+        }
+
+        consumer = testConsumer(1);
+        recordsProducer.start(consumer, blackHole);
+
+        final String toastedValue = RandomStringUtils.randomAlphanumeric(10000);
+
+        // INSERT
+        String statement = "INSERT INTO test_table (not_toast, text) VALUES (10,'" + toastedValue + "');";
+        assertInsert(
+                statement,
+                Arrays.asList(
+                        new SchemaAndValueField("id", SchemaBuilder.INT32_SCHEMA, 1), // SERIAL is NOT NULL implicitly
+                        new SchemaAndValueField("not_toast", SchemaBuilder.OPTIONAL_INT32_SCHEMA, 10),
+                        new SchemaAndValueField("text", SchemaBuilder.OPTIONAL_STRING_SCHEMA, toastedValue)
+                )
+        );
+
+        // UPDATE
+        consumer.expects(1);
+        executeAndWait("UPDATE test_table set not_toast = 20");
+        SourceRecord updatedRecord = consumer.remove();
+
+        assertRecordSchemaAndValues(Arrays.asList(
+                new SchemaAndValueField("id", SchemaBuilder.INT32_SCHEMA, 1),
+                new SchemaAndValueField("not_toast", SchemaBuilder.OPTIONAL_INT32_SCHEMA, 10),
+                new SchemaAndValueField("text", SchemaBuilder.OPTIONAL_STRING_SCHEMA, toastedValue)
+        ), updatedRecord, Envelope.FieldName.BEFORE);
+        assertRecordSchemaAndValues(Arrays.asList(
+                new SchemaAndValueField("id", SchemaBuilder.INT32_SCHEMA, 1),
+                new SchemaAndValueField("not_toast", SchemaBuilder.OPTIONAL_INT32_SCHEMA, 20)
```

Comment:
So why is the `text` field not present?

Reply:
@gunnarmorling The toasted field is not sent by the plugin.

Reply:
Understood, but does this mean we actually produce a schema which doesn't contain the value, too? That seems troublesome when using the schema registry, no? I'd have expected that only the *value* is missing, but not the field in the schema.